gunicorn/config.py

```diff
@@ -1236,10 +1236,16 @@ class SecureSchemeHeader(Setting):
     desc = """\
 
         A dictionary containing headers and values that the front-end proxy
-        uses to indicate HTTPS requests. These tell Gunicorn to set
+        uses to indicate HTTPS requests. If the source IP is permitted by
+        ``forwarded-allow-ips`` (below), *and* at least one request header matches
+        a key-value pair listed in this dictionary, then Gunicorn will set
         ``wsgi.url_scheme`` to ``https``, so your application can tell that the
         request is secure.
 
+        If the other headers listed in this dictionary are not present in the request, they will be ignored,
+        but if the other headers are present and do not match the provided values, then
+        the request will fail to parse. See the note below for more detailed examples of this behaviour.
```

Comment:
This paragraph is a little bit confusing to me. Perhaps the first sentence is unnecessary and it would be better to focus on the failure case.

"If any of the headers are present and do no match their expected value, then the request will fail."

Reply:
It's important to note that any *one* key in  `secure_scheme_headers` must match, not *every* key. That's what I'm getting at in that first sentence.

Reply:
Maybe "Requests only need to provide one header with its expected value. If requests provide more than one header, all headers must have their expected values or the request will fail."

Does something like this work?

Reply:
How about a more formalised:
> In order to enable the HTTPS behaviour:
> * At least one header in the request must be present as a key in the dictionary
> * All keys in the dictionary that are match the name of a request header must also exactly match by value

Reply:
I'm not actually certain this is better, but I haven't found any of our suggestions entirely satisfactory. Possibly because English just isn't perfect at expressing this precise condition.